writeaheadlog.go

```diff
@@ -224,15 +215,48 @@ func (w *WAL) recover(data []byte) ([]Update, error) {
 	updates := []Update{}
 	for _, sp := range firstPages {
 		var txn Transaction
-		if err = unmarshalTransaction(&txn, sp.p, sp.nextPage, data); err == nil {
-			updates = append(updates, txn.Updates...)
+		err := unmarshalTransaction(&txn, sp.p, sp.nextPage, data)
+		if err != nil {
+			continue
 		}
+		updates = append(updates, txn.Updates...)
+	}
+
+	// If there were no updates we can savely set the recovery to complete
+	if len(updates) == 0 {
+		w.recoveryComplete = true
 	}
-	w.filePageCount = uint64(len(w.availablePages))
 
 	return updates, nil
 }
 
+// RecoveryComplete is called after a wal is recovered to signal that it is
+// save to reset the wal
+func (w *WAL) RecoveryComplete() error {
+	// Marshal the pageStatusApplied
+	buffer := new(bytes.Buffer)
+	err := binary.Write(buffer, binary.LittleEndian, uint64(pageStatusApplied))
+	if err != nil {
+		return err
+	}
+
+	// Get the length of the file
+	length, err := w.logFile.Seek(0, io.SeekStart)
```

Comment:
doesn't this need to be `io.SeekEnd`?

Reply:
also, the intent is clearer if we instead use `file.Stat` instead of `Seek`

Reply:
or maybe `os.Stat(w.path)`?

Reply:
You are right. This should be io.SeekEnd.
`ogFile` is not an `os.File` and exposes only a minimal interface. It was easier to expose `Seek` instead of `file.Stat` and `fileinfo.Size`.